mava/systems/q_learning/anakin/rec_qmix.py

```diff
@@ -0,0 +1,697 @@
+    init_transition = Transition(
+        obs=init_obs,  # (A, ...)
+        action=init_acts,  # (A,)
+        reward=jnp.zeros((1,), dtype=float),
+        terminal=jnp.zeros((1,), dtype=bool),
+        term_or_trunc=jnp.zeros((1,), dtype=bool),
+        next_obs=init_obs,
+    )
+
+    # Initialise trajectory buffer
+    rb = fbx.make_trajectory_buffer(
+        # n transitions gives n-1 full data points
+        sample_sequence_length=cfg.system.sample_sequence_length,
+        period=1,  # sample any unique trajectory
+        add_batch_size=cfg.arch.num_envs,
+        sample_batch_size=cfg.system.sample_batch_size,
+        max_length_time_axis=cfg.system.buffer_size,
+        min_length_time_axis=cfg.system.min_buffer_size,
+    )
+    buffer_state = rb.init(init_transition)
+    buffer_state = replicate(buffer_state)
+
+    # Reset env
+    n_keys = cfg.arch.num_envs * cfg.arch.n_devices * cfg.system.update_batch_size
+    key_shape = (cfg.arch.n_devices, cfg.system.update_batch_size, cfg.arch.num_envs, -1)
+    key, reset_key = jax.random.split(key)
+    reset_keys = jax.random.split(reset_key, n_keys)
+    reset_keys = jnp.reshape(reset_keys, key_shape)
+
+    # Get initial state and timestep per-device
+    env_state, first_timestep = jax.pmap(  # devices
+        jax.vmap(  # update_batch_size
+            jax.vmap(env.reset),  # num_envs
+            axis_name="batch",
+        ),
+        axis_name="device",
+    )(reset_keys)
+    first_obs = first_timestep.observation
+    first_term_or_trunc = first_timestep.last()[..., jnp.newaxis]
+    first_term = (1 - first_timestep.discount[..., 0, jnp.newaxis]).astype(bool)
+
+    # Initialise env steps and training steps
+    t0_act = jnp.zeros((cfg.arch.n_devices, cfg.system.update_batch_size), dtype=int)
+    t0_train = jnp.zeros((cfg.arch.n_devices, cfg.system.update_batch_size), dtype=int)
+
+    # Keys passed to learner
+    first_keys = jax.random.split(key, (cfg.arch.n_devices * cfg.system.update_batch_size))
+    first_keys = first_keys.reshape((cfg.arch.n_devices, cfg.system.update_batch_size, -1))
+
+    # Initial learner state.
+    learner_state = LearnerState(
+        first_obs,
+        first_term,
+        first_term_or_trunc,
+        init_hidden_state,
+        env_state,
+        t0_act,
+        t0_train,
+        opt_state,
+        buffer_state,
+        params,
+        first_keys,
+    )
+
+    return (env, eval_env), q_net, q_mixer, opt, rb, learner_state, logger, key
+
+
+def make_update_fns(
+    cfg: DictConfig,
+    env: MarlEnv,
+    q_net: RecQNetwork,
+    mixer: QMixNetwork,
+    opt: optax.GradientTransformation,
+    rb: TrajectoryBuffer,
+) -> Any:
+    def select_eps_greedy_action(
+        action_selection_state: ActionSelectionState,
+        obs: Observation,
+        term_or_trunc: Array,
+    ) -> Tuple[ActionSelectionState, Array]:
+        """Select action to take in eps-greedy way. Batch and agent dims are included."""
```

Comment:
I see you removed the docstring from here, I agree with it, can you just do the same for IQL :pray: